### PR TITLE
updated applyFilters 

### PR DESCRIPTION
**Pull request by drnickisaac**
I also added tempTrendSummaries, for security

### CODE REVIEW

**Comment by AugustT** [comment]
If this takes time we should add a logical `parallel` arguement which if `TRUE` the lapply becomes a `mclapply`

**Comment by AugustT** [comment]
Datalabs RStudio server sessions have 8 cores, so this would provide a ~4 times speed increase

File: R/tempTrendSummaries.R [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+      #      firstMod <- tn
+      #    }
+      
+      out2 <- data.frame(species, nRec, first, last, gap, firstMod, lastMod)
+      return(list(out1, out2))
+    } else return(NULL)
+  }
+  
+  if(parallel) outputs <- parallel::mclapply(spp.list, 
+                    combineSamps, minObs=minObs, mc.cores = 4)
```
Comment (AugustT) [inline R/tempTrendSummaries.R:104]: @drnickisaac @robboyd you shouldn't hard code the number of cores. Change 4 to parallel::detect.cores()-1 as default but allow user to choose

Comment (drnickisaac) [reply R/tempTrendSummaries.R:104]: Ok - thanks @AugustT . I have edited me local version and will suggest merge in due course.

**Pull request merged**